libphodav/chezdav.c

```diff
@@ -228,12 +232,21 @@ main (int argc, char *argv[])
     my_error (_ ("option parsing failed: %s\n"), error->message);
   g_option_context_free (context);
 
+  if (argc != 1)
+    my_error (_ ("unsupported extra arguments: %s ...\n"), argv[1]);
+
   if (version)
     {
       g_printf (PACKAGE_STRING "\n");
       return 0;
     }
 
+  if (local && public)
+    my_error (_ ("--local and --public are mutually exclusive\n"));
+
+  if (!local && !public)
+    public = 1; // default
```

Comment:
note that this changes the default to public, which might not be desireable,
as it might make user's existing setups less secure.